Files.Launcher/MessageHandlers/FileOperationsHandler.cs

```diff
@@ -0,0 +1,540 @@
+                            op.PostDeleteItem += (s, e) =>
+                            {
+                                if (e.Result.Succeeded)
+                                {
+                                    if (!fileToDeletePath.Any(x => x == e.SourceItem.FileSystemPath))
+                                    {
+                                        return;
+                                    }
+                                    deletedItems.Add(e.SourceItem.FileSystemPath);
+                                    if (e.DestItem != null)
+                                    {
+                                        recycledItems.Add(e.DestItem.FileSystemPath);
+                                    }
+                                }
+                            };
+                            op.FinishOperations += (s, e) => deleteTcs.TrySetResult(e.Result.Succeeded);
+                            op.UpdateProgress += async (s, e) => await Win32API.SendMessageAsync(connection, new ValueSet() {
+                                { "Progress", e.ProgressPercentage },
+                                { "OperationID", operationID2 }
+                            });
+                            op.UpdateProgress += (s, e) =>
+                            {
+                                if (handleTable.GetValue<bool>(operationID2))
+                                {
+                                    throw new Win32Exception(unchecked((int)0x80004005)); // E_FAIL, stops operation
+                                }
+                            };
+
+                            try
+                            {
+                                op.PerformOperations();
+                            }
+                            catch
+                            {
+                                deleteTcs.TrySetResult(false);
+                            }
+
+                            handleTable.RemoveValue(operationID2);
+
+                            return (await deleteTcs.Task && deletedItems.Count == fileToDeletePath.Length, deletedItems, recycledItems);
+                        }
+                    });
+                    await Win32API.SendMessageAsync(connection, new ValueSet() {
+                        { "Success", succcess2 },
+                        { "DeletedItems", JsonConvert.SerializeObject(deletedItems) },
+                        { "RecycledItems", JsonConvert.SerializeObject(recycledItems) }
+                    }, message.Get("RequestID", (string)null));
+                    break;
+
+                case "RenameItem":
+                    var fileToRenamePath = (string)message["filepath"];
+                    var newName = (string)message["newName"];
+                    var operationID4 = (string)message["operationID"];
+                    var overwriteOnRename = (bool)message["overwrite"];
+                    var (succcess4, renamedItems) = await Win32API.StartSTATask(async () =>
```

Comment:
To avoid having to add numbers to multiple variables of the same name in switch statement, you could wrap every `case` in brackets, like so:

```cs
switch()
{
    case 1:
    {
        int success = 0;
        break;
    }
    case 2:
    {
        int success = 0;
        break;
    }
}

```